soundata/annotations.py

```diff
@@ -36,39 +64,130 @@ class Events(Annotation):
         validate_array_like(intervals, np.ndarray, float)
         validate_array_like(labels, list, str)
         validate_array_like(confidence, np.ndarray, float, none_allowed=True)
         validate_lengths_equal([intervals, labels, confidence])
         validate_intervals(intervals)
         validate_confidence(confidence)
+        validate_unit(labels_unit, LABEL_UNITS)
+        validate_unit(intervals_unit, TIME_UNITS)
 
         self.intervals = intervals
+        self.intervals_unit = intervals_unit
         self.labels = labels
+        self.labels_unit = labels_unit
         self.confidence = confidence
 
 
+class SpatialEvents(Events):
+    """SpatialEvents class
+
+    Attributes:
+        intervals (np.ndarray): (n x 2) array of intervals
+            (as floats) in seconds in the form [start_time, end_time]
+            with positive time stamps and end_time >= start_time.
+        labels (list): list of event labels (as strings)
+        confidence (np.ndarray or None): array of confidence values
+        labels_unit (str): labels unit, one of LABELS_UNITS
+        intervals_unit (str): intervals unit, one of TIME_UNITS
+        azimuth (np.ndarray or None):
+        azimuth_unit (str): azimuth unit, one of AZIMUTH_UNITS
+        elevation (np.ndarray or None):
+        elevation_unit (str): elevation unit, one of AZIMUTH_UNITS
+        distance (np.ndarray or None):
+        distance_unit (str): distance unit, one of DISTANCE_UNITS
+        cartesian_coord (np.ndarray or None):
+        cartesian_coord_unit (str): cartesian_coord unit, one of DISTANCE_UNITS
+
+
+    """
+
+    def __init__(
+        self,
+        intervals,
+        intervals_unit,
+        labels,
+        labels_unit,
+        confidence=None,
+        azimuth=None,
+        azimuth_unit=None,
+        elevation=None,
+        elevation_unit=None,
+        distance=None,
+        distance_unit=None,
+        cartesian_coord=None,
+        cartesian_coord_unit=None,
+    ) -> None:
+
+        Events.__init__(
+            self, intervals, intervals_unit, labels, labels_unit, confidence=confidence
+        )
+
+        validate_array_like(azimuth, np.ndarray, float)
+        validate_array_like(elevation, np.ndarray, float)
+        validate_array_like(distance, np.ndarray, float)
+        validate_array_like(cartesian_coord, np.ndarray, float)
+        validate_lengths_equal([azimuth, elevation, distance, cartesian_coord])
+        validate_azimuth(azimuth, azimuth_unit, allow_none=True)
+        validate_distance(distance, allow_none=True)
+        validate_azimuth(elevation, elevation_unit, allow_none=True)
+        validate_cartesian_coord(cartesian_coord, allow_none=True)
+        validate_unit(azimuth_unit, AZIMUTH_UNITS, allow_none=True)
+        validate_unit(distance_unit, DISTANCE_UNITS, allow_none=True)
+        validate_unit(elevation_unit, AZIMUTH_UNITS, allow_none=True)
+        validate_unit(cartesian_coord_unit, DISTANCE_UNITS, allow_none=True)
```

Comment:
per previous comment, should validation also compare these time series to the interval to ensure they match? Should the hop size be explicitly stored somewhere for these time series?

Reply:
The time-step can be part of the metadata. I think that would be the best location.